### PR TITLE
Media API Elastic6 toggle

### PR DESCRIPTION
**Pull request by tonytw1**
Media API can route requests to ES1 or ES6 based on the presences of a feature toggle cookie.

### CODE REVIEW

File: media-api/app/controllers/AggregationController.scala [hunk 1]
```diff
@@ -11,6 +11,8 @@ class AggregationController(auth: Authentication, elasticSearch: ElasticSearchVe
   extends BaseController with AggregateResponses {
 
   def dateHistogram(field: String, q: Option[String]) = auth.async { request =>
+    implicit val r = request
```
Comment (mbarton) [inline media-api/app/controllers/AggregationController.scala]: I think you can inline this above:

```scala
  def dateHistogram(field: String, q: Option[String]) = auth.async { implicit request =>
```

File: media-api/app/lib/UsageQuota.scala [hunk 2]
```diff
@@ -37,7 +40,7 @@ class UsageQuota(config: MediaApiConfig, elasticSearch: ElasticSearchVersion, sc
       waitMillis.millis)
   }.toOption.exists(_.exceeded) && FeatureToggle.get("usage-quota-ui")
 
-  def usageStatusForImage(id: String): Future[UsageStatus] = for {
+  def usageStatusForImage(id: String)(implicit request: AuthenticatedRequest[AnyContent, Principal]): Future[UsageStatus] = for {
```
Comment (mbarton) [inline media-api/app/lib/UsageQuota.scala]: Any reason for this to be an implicit parameter rather than an explicit one?

Comment (tonytw1) [reply media-api/app/lib/UsageQuota.scala]: Minimising the change and preserving the explicit signature which was already there as this request awareness is a temporary thing.

File: media-api/app/controllers/AggregationController.scala (hunk 1 above)
Comment (tonytw1) [reply media-api/app/controllers/AggregationController.scala]: Probably; if none of the downstream code already references the explicit request; hence I'm providing both.

**Pull request merged**